### PR TITLE
Sanitize tag, item, and mod inputs

### LINKED ISSUE #53: OreDict entry creation has no input validation
The creation functionality does not perform any checks against the data provided so you can literally put nothing and it will create an entry.

### PR DESCRIPTION
**Pull request by elifoster**
Close #53.

@Alexia Could you review this and perhaps double check that I haven't missed any areas where input sanitation is needed?

### CODE REVIEW

File: special/OreDictEntryManager.php [hunk 1]
```diff
@@ -99,9 +116,20 @@ private function createEntry(FormOptions $opts) {
 			return -2;
 		}
 
-		return OreDict::addEntry($mod, $item, $tag, $this->getUser(), $params);
+		$result = OreDict::addEntry($mod, $item, $tag, $this->getUser(), $params);
+		if ($result == false) {
+			return -2;
+		}
+		return $result;
```
Comment (xbony2) [inline special/OreDictEntryManager.php]: Is ? : a thing in PHP?

Comment (Alexia) [inline special/OreDictEntryManager.php]: Also loosely checking against false in PHP, while rare, could end in disaster.

Comment (xbony2) [inline special/OreDictEntryManager.php]: Come to think about it, in Java the syntax would be !$results although I suspect that's not a possible thing in PHP.

Comment (Alexia) [inline special/OreDictEntryManager.php]:
```
return ($result === false ? -2 : $result);
```

**Comment by Alexia** [comment]
This seems fine to me.

File: special/OreDictEntryManager.php (hunk 1 above)
Comment (elifoster) [inline special/OreDictEntryManager.php]: @Alexia Is that check loose? I thought `=== false` would be "loose" while `== false` would be strictly checking against `false`.

`OreDict#addEntry` returns either `false` or the entry ID. Is this a poor way to handle this?

**Pull request merged**